PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionContentPanel.java

```diff
@@ -565,71 +475,138 @@ private void addCommandlineArgsLineItem() {
+    }
+
+    private void cleanUp() {
         clusterSelectedSubject.onCompleted();
     }
+
+    private synchronized void checkInputsWithErrorLabels() {
+        // Clean all error messages firstly
+        hideAllErrors();
+
+        // Check Cluster selection
+        if (clustersListComboBox.getComboBox().getSelectedItem() == null) {
+            setVisibleForFixedErrorMessageLabel(ErrorMessage.ClusterName, true);
+        }
+
+        if (intelliJArtifactRadioButton.isSelected()) {
+            // Check Intellij artifact
+            if (selectedArtifactComboBox.getSelectedItem() == null) {
+                setVisibleForFixedErrorMessageLabel(ErrorMessage.SystemArtifact, true);
+            }
+        }
+
+        if (localArtifactRadioButton.isSelected()) {
+            // Check local jar artifact
+            if (StringHelper.isNullOrWhiteSpace(localArtifactTextField.getText())) {
+                setVisibleForFixedErrorMessageLabel(ErrorMessage.LocalArtifact, true);
+            }
+
+            if (!SparkSubmitHelper.isLocalArtifactPath(localArtifactTextField.getText())) {
+                setVisibleForFixedErrorMessageLabel(ErrorMessage.LocalArtifact, true);
+            }
+        }
+
+        // Check main class input
+        if (StringUtils.isBlank(mainClassTextField.getText())) {
+            setVisibleForFixedErrorMessageLabel(ErrorMessage.MainClass, true);
+        }
+
+        // Check job config table
+        SubmissionTableModel confTableModel = ((SubmissionTableModel) getJobConfigurationTable().getModel());
+        SparkSubmissionJobConfigCheckResult result = confTableModel.getFirstCheckResults();
+        if (result != null) {
+            setStatusForMessageLabel(ErrorMessage.JobConfiguration, true, result.getMessaqge(), result.getStatus() == SparkSubmissionJobConfigCheckStatus.Warning);
+        }
+    }
+
+    public void checkInputs() throws ConfigurationException {
+        checkInputsWithErrorLabels();
+
+        // Convert Error Labels into Configuration Exception
+        java.util.List<String> errors = getErrorMessages();
+        if (!errors.isEmpty()) {
+            throw new ConfigurationException(String.join("; \n", errors));
+        }
+
+        SubmissionTableModel confTableModel = ((SubmissionTableModel) getJobConfigurationTable().getModel());
+        for (Pair<String, String> confEntry : confTableModel.getJobConfigMap()) {
+            String entryKey = confEntry.first();
+
+            if (!StringUtils.isAlpha(entryKey.substring(0, 1)) && !StringUtils.startsWith(entryKey, "_")) {
+                throw new RuntimeConfigurationError("The Spark config key should start with a letter or underscore");
+            }
+
+            if (!StringUtils.containsOnly(entryKey.toLowerCase(), "abcdefghijklmnopqrstuvwxyz1234567890_-.")) {
+                throw new RuntimeConfigurationError("The Spark config key should only contains letters, digits, hyphens, underscores, and periods: (" + entryKey + ")");
```

Comment:
“... should only contains lower case letters, digits, ...”

Reply:
I've converted the key to lower case to check 😃